packages/web-app-files/src/views/LocationPicker.vue

```diff
@@ -460,13 +476,27 @@ export default {
                   targetPath
                 )
                 break
+              case 'personal':
               default:
-                targetPath = buildWebDavFilesPath(this.user.id, this.target || '/')
-                targetPath += `/${resourceName}`
-                promise = this.$client.files.copy(
-                  buildWebDavFilesPath(this.user.id, resource),
-                  targetPath
-                )
+                if (this.hasShareJail) {
+                  targetPath = buildWebDavSpacesPath(
+                    this.$route.query.storageId,
+                    this.target || '/'
+                  )
+                  resourceName = basename(resource)
+                  targetPath += `/${resourceName}`
+                  promise = this.$client.files.copy(
+                    buildWebDavSpacesPath(this.$route.query.storageId, resource),
+                    targetPath
+                  )
+                } else {
+                  targetPath = buildWebDavFilesPath(this.user.id, this.target || '/')
+                  targetPath += `/${resourceName}`
+                  promise = this.$client.files.copy(
+                    buildWebDavFilesPath(this.user.id, resource),
+                    targetPath
+                  )
+                }
```

Comment:
Can we reduce the code duplication here? This block is almost exactly the same as for the move operation.